src/main/java/org/dspace/pack/bagit/BagItAipReader.java

```diff
@@ -63,16 +76,44 @@ public BagItAipReader(final Path bag) throws IOException {
             throw new IOException("Missing archive: " + bag);
         }
 
+        // get the BagProfile
+        final ConfigurationService configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        final String profileName = configurationService.getProperty(BAG_PROFILE_KEY, DEFAULT_PROFILE);
+        this.profile = new BagProfile(BagProfile.BuiltIn.from(profileName));
+
         // deserialize if necessary
         if (Files.isRegularFile(bag)) {
-            final BagProfile profile = new BagProfile(BagProfile.BuiltIn.BEYOND_THE_REPOSITORY);
             final BagDeserializer deserializer = SerializationSupport.deserializerFor(bag, profile);
             this.bag = deserializer.deserialize(bag);
         } else {
             this.bag = bag;
         }
     }
 
+    /**
+     * Validate that an AIP is in a BagIt format which passes both bagit-bag validation and bagit-profile validation
+     *
+     * @throws RuntimeException if there is an error during validation
+     */
+    public void validateBag() {
+        final Bag locBag;
+        final BagReader bagReader = new BagReader();
+        final BagVerifier verifier = new BagVerifier() ;
+        try {
+            locBag = bagReader.read(bag);
+        } catch (UnparsableVersionException | InvalidBagitFileFormatException | UnsupportedAlgorithmException
+            | MaliciousPathException | IOException e) {
+            throw new RuntimeException("Unable to read aip as a BagIt bag!", e);
+        }
+
+        try {
+            profile.validateBag(locBag);
+            verifier.isValid(locBag, false);
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to verify BagIt bag!\n" + e.getMessage());
```

Comment:
Rather than just printing `e.getMessage()`, we might want to treat this the same as the other RuntimeException in this method, i.e.
`throw new RuntimeException("Unable to verify BagIt bag!", e);`